server/src/main/java/io/spine/server/entity/RecentHistory.java

```diff
@@ -0,0 +1,118 @@
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import io.spine.core.Event;
+
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.Objects;
+
+import static com.google.common.collect.Queues.newArrayDeque;
+
+/**
+ * Recent history of an {@linkplain io.spine.server.entity.EventPlayingEntity event-sourced entity}.
+ *
+ * @author Mykhailo Drachuk
+ * @author Alexander Yevsyukov
+ */
+public final class RecentHistory {
+
+    /**
+     * Holds the history of all events which happened to the aggregate since the last snapshot.
+     *
+     * <p>Most recent event come first.
+     *
+     * @see #iterator()
+     */
+    private final Deque<Event> history = newArrayDeque();
+
+    /**
+     * Creates new instance.
+     */
+    RecentHistory() {
+        super();
+    }
+
+    /**
+     * Returns {@code true} if there are no events in the recent history, {@code false} otherwise.
+     */
+    public boolean isEmpty() {
+        return history.isEmpty();
+    }
+
+    /**
+     * Removes all events from the recent history.
+     */
+    void clear() {
+        history.clear();
+    }
+
+    /**
+     * Creates a new iterator over the recent history items.
+     *
+     * <p>The iterator returns events in the reverse chronological order. That is, most recent
+     * event would be returned first.
+     *
+     * @return an events iterator
+     */
+    public Iterator<Event> iterator() {
+        final ImmutableList<Event> events = ImmutableList.copyOf(history);
+        return events.iterator();
+    }
+
+    /**
+     * Adds events to the aggregate history.
+     *
+     * @param events events in the chronological order
+     */
+    void addAll(Iterable<Event> events) {
+        for (Event event : events) {
+            history.addFirst(event);
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(history);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        final RecentHistory other = (RecentHistory) obj;
+        return Objects.equals(this.history, other.history);
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+                          .add("history", history)
```

Comment:
This implementation does not seem useful, as there may be dozens of those. I would print the number of events and that's it.

Reply:
Yes, you're right. 

I've submitted an issue for possible extension of this output in the future (#679).